### PR TITLE
migration

### PR DESCRIPTION
**Pull request by fdaei**
plz

### CODE REVIEW

**Comment by fdaei** [comment]
plz

File: src/migrations/m230517_063258_widget_table.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     */
+    public function safeUp()
+    {
+        $tableOptions = null;
+        if ($this->db->driverName === 'mysql') {
+            $tableOptions = 'CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci ENGINE=InnoDB';
+        }
+        $this->createTable('{{%widget}}', [
+            'id' => $this->primaryKey()->unsigned()->notNull(),
+            'title' => $this->string(255)->notNull(),
```
Comment (Sadi01) [inline src/migrations/m230517_063258_widget_table.php]: 128 char is enough

File: src/migrations/m230517_063258_widget_table.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function safeUp()
+    {
+        $tableOptions = null;
+        if ($this->db->driverName === 'mysql') {
+            $tableOptions = 'CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci ENGINE=InnoDB';
+        }
+        $this->createTable('{{%widget}}', [
```
Comment (Sadi01) [inline src/migrations/m230517_063258_widget_table.php]: report_widget

File: src/migrations/m230517_063258_widget_table.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            'id' => $this->primaryKey()->unsigned()->notNull(),
+            'title' => $this->string(255)->notNull(),
+            'description' => $this->string(255)->notNull(),
+            'search_model' => $this->string(255)->notNull(),
```
Comment (Sadi01) [inline src/migrations/m230517_063258_widget_table.php]: search_model_class

type smallint, unsigned

File: src/migrations/m230517_063258_widget_table.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            'method' => $this->string(255)->notNull(),
```
Comment (Sadi01) [inline src/migrations/m230517_063258_widget_table.php]: search_model_method
128 char is enough

File: src/migrations/m230517_063258_widget_table.php [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            'status' => $this->tinyInteger()->notNull()->defaultValue(0),
```
Comment (Sadi01) [inline src/migrations/m230517_063258_widget_table.php]: default value is 1

File: src/migrations/m230517_063258_widget_table.php [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            'deleted_at' => $this->integer()->unsigned()->notNull(),
```
Comment (Sadi01) [inline src/migrations/m230517_063258_widget_table.php]: set default value to 0

File: src/migrations/m230517_063258_widget_table.php [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+            'updated_at' => $this->integer()->unsigned()->notNull(),
+            'created_at' => $this->integer()->unsigned()->notNull(),
+            'updated_by' => $this->integer()->unsigned()->notNull(),
+            'created_by' => $this->integer()->unsigned()->notNull(),
+        ], $tableOptions);
```
Comment (Sadi01) [inline src/migrations/m230517_063258_widget_table.php]: add 4 columns as below:

search_model_run_result_view ==> varchar(128)
range_type ==> tinyint, unsigned
visibility ==> tinyint, unsigned
add_on ==> json, defaulValue ==> json_object()

File: src/migrations/m230517_064501_result_widget_table.php [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+class m230517_064501_result_widget_table extends Migration
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function safeUp()
+    {
+        $this->createTable('{{%widget_result}}', [
+            'id' => $this->primaryKey()->unsigned(),
+            'result' => $this->json(),
```
Comment (Sadi01) [inline src/migrations/m230517_064501_result_widget_table.php]: change result to add_on, defaultValue ==> json_object()

File: src/migrations/m230517_064501_result_widget_table.php [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+            'widget_id' => $this->integer()->unsigned()->notNull(),
+            'start_at' => $this->integer()->unsigned()->notNull(),
```
Comment (Sadi01) [inline src/migrations/m230517_064501_result_widget_table.php]: start_range

File: src/migrations/m230517_064501_result_widget_table.php [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+            'end_at' => $this->integer()->unsigned()->notNull(),
```
Comment (Sadi01) [inline src/migrations/m230517_064501_result_widget_table.php]: end_range

File: src/migrations/m230517_064501_result_widget_table.php [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+            'status' => $this->integer()->unsigned()->notNull(),
+            'parameters' => $this->json(),
```
Comment (Sadi01) [inline src/migrations/m230517_064501_result_widget_table.php]: remove parameters column

File: src/migrations/m230517_064501_result_widget_table.php [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+            'updated_at' => $this->integer()->unsigned()->notNull(),
+            'created_at' => $this->integer()->unsigned()->notNull(),
+            'deleted_at' => $this->integer()->unsigned()->notNull(),
```
Comment (Sadi01) [inline src/migrations/m230517_064501_result_widget_table.php]: Set dafaultValue  to 0

File: src/migrations/m230517_064501_result_widget_table.php [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+     */
+    public function safeUp()
+    {
+        $this->createTable('{{%widget_result}}', [
+            'id' => $this->primaryKey()->unsigned(),
+            'result' => $this->json(),
+            'widget_id' => $this->integer()->unsigned()->notNull(),
+            'start_at' => $this->integer()->unsigned()->notNull(),
+            'end_at' => $this->integer()->unsigned()->notNull(),
+            'status' => $this->integer()->unsigned()->notNull(),
```
Comment (Sadi01) [inline src/migrations/m230517_064501_result_widget_table.php]: Set defalutValue to 1

File: src/migrations/m230517_064501_result_widget_table.php [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+            'parameters' => $this->json(),
+            'updated_at' => $this->integer()->unsigned()->notNull(),
+            'created_at' => $this->integer()->unsigned()->notNull(),
+            'deleted_at' => $this->integer()->unsigned()->notNull(),
+            'update_by' => $this->integer()->unsigned()->notNull(),
+            'created_by' => $this->integer()->unsigned()->notNull(),
```
Comment (Sadi01) [inline src/migrations/m230517_064501_result_widget_table.php]: add 2 columns as below:

run_controller ==> varchar(256)
run_action ==> varchar(128)

File: src/migrations/m230517_065155_page_table.php [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+     */
+    public function safeUp()
+    {
+        $this->createTable('{{%page}}', [
+            'id' => $this->primaryKey()->unsigned(),
+            'title' => $this->string(255)->notNull(),
+            'route' => $this->string(255)->notNull(),
+            'created_at' => $this->integer()->unsigned()->notNull(),
+            'updated_at' => $this->integer()->unsigned()->notNull(),
+            'deleted_at' => $this->integer()->unsigned(),
```
Comment (Sadi01) [inline src/migrations/m230517_065155_page_table.php]: Set defaultValue to 0

File: src/migrations/m230517_065155_page_table.php [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function safeUp()
+    {
+        $this->createTable('{{%page}}', [
+            'id' => $this->primaryKey()->unsigned(),
+            'title' => $this->string(255)->notNull(),
+            'route' => $this->string(255)->notNull(),
```
Comment (Sadi01) [inline src/migrations/m230517_065155_page_table.php]: Remove route column

File: src/migrations/m230517_065155_page_table.php [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+            'created_at' => $this->integer()->unsigned()->notNull(),
+            'updated_at' => $this->integer()->unsigned()->notNull(),
+            'deleted_at' => $this->integer()->unsigned(),
+            'updated_by' => $this->integer()->unsigned(),
+            'created_by' => $this->integer()->unsigned(),
+        ]);
```
Comment (Sadi01) [inline src/migrations/m230517_065155_page_table.php]: add 2 columns as below:
status ==> tinyint, defaultValue ==> 1
range_type ==> tinyint
add_on ==> json, defaultValue ==> json_object()

File: src/migrations/m230517_065155_page_table.php [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function safeDown()
+    {
+        echo "m230517_065155_page_table cannot be reverted.\n";
```
Comment (Sadi01) [inline src/migrations/m230517_065155_page_table.php]: Write safeDown method

File: src/migrations/m230517_065155_page_table.php [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+
+        return false;
+    }
+
+    /*
```
Comment (Sadi01) [inline src/migrations/m230517_065155_page_table.php]: remove unused commented codes

File: src/migrations/m230517_065160_create_page_widget_table.php [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function safeUp()
+    {
+        $this->createTable('{{%page_widget}}', [
+            'id' => $this->primaryKey()->unsigned(),
+            'page_id' => $this->integer()->unsigned()->notNull(),
+            'widget_id' => $this->integer()->unsigned()->notNull(),
+            'shape_id' => $this->integer()->unsigned()->notNull(),
```
Comment (Sadi01) [inline src/migrations/m230517_065160_create_page_widget_table.php]: remove shape_id column

File: src/migrations/m230517_065160_create_page_widget_table.php [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+            'created_at' => $this->integer()->unsigned()->notNull(),
+            'updated_at' => $this->integer()->unsigned()->notNull(),
+            'deleted_at' => $this->integer()->unsigned(),
+            'updated_by' => $this->integer()->unsigned(),
+            'created_by' => $this->integer()->unsigned(),
+        ]);
```
Comment (Sadi01) [inline src/migrations/m230517_065160_create_page_widget_table.php]: add 2 columns as below:

report_widget_field ==> varchar(64)
report_widget_field_format ==> tinyint
status ==> tinyint, defalutValue ==> 1
deleted_at ==> int unsigned, defaultVlaue ==> 0

**Review by Sadi01 — APPROVED** [review]

**Pull request merged**